### PR TITLE
Update Flutter to use the latest dartdoc

### PR DESCRIPTION
**Pull request by eseidelGoogle**
Now that https://github.com/dart-lang/dartdoc/issues/1236 is fixed.

This can't be merged until dartdoc does a release.

### CODE REVIEW

**Comment by eseidelGoogle** [comment]
@devoncarew

**Comment by eseidelGoogle** [comment]
https://pub.dartlang.org/packages/dartdoc has been updated, this can be merged.  @Hixie

**Comment by devoncarew** [comment]
lgtm

We may want to do some light-weight validation of the generated output as well (some file existence checks, a regexs of some content, ...).

**Comment by Hixie** [comment]
This should have a test.

File: dev/tools/dartdoc.dart [hunk 1]
```diff
@@ -62,16 +62,14 @@ dependencies:
     'global', 'run', 'dartdoc',
     '--header', 'styles.html',
     '--header', 'analytics.html',
-    '--dart-sdk', '../../bin/cache/dart-sdk',
```
Comment (Hixie) [inline dev/tools/dartdoc.dart]: can you elaborate on why this change?

Comment (eseidelGoogle) [inline dev/tools/dartdoc.dart]: That argument was removed from dartdoc.

**Comment by abarth** [comment]
I'm going to steal this from @eseidelGoogle

**Comment by abarth** [comment]
Updated PR in https://github.com/flutter/flutter/pull/7438
